### PR TITLE
Move anaconda-specific code to anaconda

### PR DESCRIPTION
**Pull request by dwlehman**

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

File: blivet/formats/disklabel.py [hunk 1]
```diff
@@ -218,12 +218,71 @@ def parted_device(self):
...
+    @classmethod
+    def set_default_label_type(cls, labeltype):
+        cls._default_label_type = labeltype
+        log.debug("default disklabel has been set to %s", labeltype)
+
+    def _label_type_size_check(self, label_type):
+        if self.parted_device is None:
+            return False
+
+        _label = parted.freshDisk(device=self.parted_device, ty=label_type)
```
Comment (vpodzime) [inline blivet/formats/disklabel.py]: This should be just ``label``

File: blivet/formats/disklabel.py [hunk 2]
```diff
@@ -218,12 +218,71 @@ def parted_device(self):
...
+        return self.parted_device.length < _label.maxPartitionStartSector
+
+    def _get_best_label_type(self):
+        label_type = self.__class__._default_label_type
+        label_types = self.__class__.get_platform_label_types()[:]
```
Comment (vpodzime) [inline blivet/formats/disklabel.py]: The ``__class__`` part here can be dropped, AFAICT.

File: blivet/formats/disklabel.py [hunk 3]
```diff
@@ -218,12 +218,71 @@ def parted_device(self):
...
+            elif self.parted_device.type == parted.DEVICE_DASD:
+                # the device is DASD
+                return "dasd"
+
+        for lt in label_types:
+            if self._label_type_size_check(lt):
+                log.debug("selecting %s disklabel for %s based on size",
+                          label_type, os.path.basename(self.device))
+                label_type = lt
+                break
```
Comment (vpodzime) [inline blivet/formats/disklabel.py]: Why not just ``return lt`` here?

File: blivet/formats/disklabel.py [hunk 4]
```diff
@@ -218,12 +218,71 @@ def parted_device(self):
...
+
+        return label_type
```
Comment (vpodzime) [inline blivet/formats/disklabel.py]: Do we really want to return the default label type even if it doesn't match the *size_check()* criteria?

File: blivet/formats/disklabel.py (hunk 1 above)
Comment (dwlehman) [reply blivet/formats/disklabel.py]: Fixed.

File: blivet/formats/disklabel.py (hunk 4 above)
Comment (dwlehman) [reply blivet/formats/disklabel.py]: We always want to return a label type. Even if none of the types pass the size test, we still need to return one.

File: blivet/formats/disklabel.py (hunk 3 above)
Comment (dwlehman) [reply blivet/formats/disklabel.py]: I just prefer to have one return statement per method, aside from very-early returns due to precondition checking.

Comment (dwlehman) [reply blivet/formats/disklabel.py]: I see I returned early in the previous s390-specific block just above this one. It was just too much change to try to cram that extremely platform-specific logic into the single-return pattern.

**Comment by vpodzime** [comment]
Still looks good to me.

**Pull request merged**